server/pxf-service/src/main/java/org/greenplum/pxf/service/HttpRequestParser.java

```diff
@@ -77,6 +91,14 @@ public RequestContext parseRequest(MultiValueMap<String, String> requestHeaders,
 
         // fill the Request-scoped RequestContext with parsed values
 
+        context.setClientApiVersion(
+                params.removeProperty("PXF-API-VERSION", ERROR_MESSAGE_HINT));
+
+        if (!apiVersionChecker.isCompatible(context.getClientApiVersion(), getServerApiVersion())) {
+            throw new PxfRuntimeException(
+                    String.format(ERROR_MESSAGE_TEMPLATE, getServerApiVersion(), context.getClientApiVersion()));
```

Comment:
let's add the hint to the exception here:
```suggestion
            throw new PxfRuntimeException(
                    String.format(ERROR_MESSAGE_TEMPLATE, getServerApiVersion(), context.getClientApiVersion()), ERROR_MESSAGE_HINT);
```